ui/server/src/main/scala/pl/touk/nussknacker/ui/process/repository/DBFetchingProcessRepository.scala

```diff
@@ -202,68 +177,53 @@ abstract class DBFetchingProcessRepository[F[_]: Monad](val dbConfig: DbConfig)
                                                                           (implicit loggedUser: LoggedUser, ec: ExecutionContext) = {
     val id = processVersion.processId
     for {
-      subprocessesVersions <- OptionT.liftF[DB, Map[ProcessName, LocalDateTime]](subprocessLastModificationDates)
       process <- OptionT[DB, ProcessEntityData](processTableFilteredByUser.filter(_.id === id).result.headOption)
-      processVersions <- OptionT.liftF[DB, Seq[ProcessVersionEntityData]](latestProcessVersions(ProcessId(id))(ProcessShapeFetchStrategy.NotFetch).result)
-      latestDeployedVersionsPerEnv <- OptionT.liftF[DB, Seq[DeployedProcessVersionEntityData]](latestDeployedProcessVersionsPerEnvironment(id).result)
+      processVersions <- OptionT.liftF[DB, Seq[ProcessVersionEntityData]](fetchProcessLatestVersions(ProcessId(id))(ProcessShapeFetchStrategy.NotFetch).result)
+      deployments <- OptionT.liftF[DB, Seq[DeployedProcessInfoEntityData]](processDeploymentsInfoPerEnvironment(id).result)
       tags <- OptionT.liftF[DB, Seq[TagsEntityData]](tagsTable.filter(_.processId === process.id).result)
     } yield createFullDetails(
       process = process,
-      processVersion = processVersion,
-      isLatestVersion = isLatestVersion,
-      currentlyDeployedAt = latestDeployedVersionsPerEnv,
+      currentVersion = processVersion,
+      initialVersion = processVersions.headOption.getOrElse(throw new  IllegalStateException("Process without version shouldn't exist")),
+      deployments = deployments,
       tags = tags,
-      history = processVersions.map(pvs => ProcessRepository.toProcessHistoryEntry(process, pvs, latestDeployedVersionsPerEnv.toList)),
-      businessView = businessView,
-      subprocessesVersions = subprocessesVersions
+      history = processVersions.map(pvs => ProcessRepository.toProcessHistoryEntry(process, pvs, deployments.toList)),
+      isLatestVersion = isLatestVersion,
+      businessView = businessView
     )
   }
 
+  //TODO: Replace initialVersion by migration createdBy and createdAt to process instance
   private def createFullDetails[PS: ProcessShapeFetchStrategy](process: ProcessEntityData,
-                                                               processVersion: ProcessVersionEntityData,
-                                                               isLatestVersion: Boolean,
```

Comment:
Was there some situation when `isLatestVersion` was `false`?

Reply:
There is situation in `fetchProcessDetailsForVersion` method when we set: `  isLatestVersion = isLatestVersion`.

Reply:
My mistake I thought that you removed this boolean flag but now I see that you just put it on the bottom. But I have other question then ... if you changed `processVersion` to `currentVersion`, this name will be invalid in described by you scenario, am I right? Because someone ask for some specific version and you will return it in `currentVersion` but in fact it won't be current but some version that was demand... Given to that previous name is right and `isLatestVersion` should be really close to this parameter because they are connected.

Reply:
You are right - reverted to old name: `processVersion`.